CHANGELOG.md

```diff
@@ -1,5 +1,75 @@
+- Support frozen dataclasses ([#17974](https://github.com/astral-sh/ruff/pull/17974))
+- Use `__getattribute__` to lookup unknown members on a type ([#18280](https://github.com/astral-sh/ruff/pull/18280))
+- Add a subdiagnostic if `invalid-return-type` is emitted on a method with an empty body on a non-protocol subclass of a protocol class ([#18243](https://github.com/astral-sh/ruff/pull/18243))
+- Do not carry the generic context of `Protocol` or `Generic` in the `ClassBase` enum ([#17989](https://github.com/astral-sh/ruff/pull/17989))
+- Improve `invalid-type-form` diagnostic where a module-literal type is used in a type expression and the module has a member which would be valid in a type expression ([#18244](https://github.com/astral-sh/ruff/pull/18244))
+- Split `invalid-base` error code into two error codes ([#18245](https://github.com/astral-sh/ruff/pull/18245))
+- Rename `call-possibly-unbound-method` to `possibly-unbound-implicit-call` ([#18017](https://github.com/astral-sh/ruff/pull/18017))
+
+### Configuration
+
+- Add `tests` to `src.root` if it exists and is not a package ([#18286](https://github.com/astral-sh/ruff/pull/18286))
+- Tell the user why we inferred the Python version we inferred ([#18082](https://github.com/astral-sh/ruff/pull/18082))
+- Resolving Python path using `CONDA_PREFIX` variable to support Conda and Pixi ([#18267](https://github.com/astral-sh/ruff/pull/18267))
+- Move `respect-ignore-files` under `src` section ([#18322](https://github.com/astral-sh/ruff/pull/18322))
+
+### Server
+
+- Fix server panic when calling `system_mut` ([#18252](https://github.com/astral-sh/ruff/pull/18252))
+- Abort process if worker thread panics ([#18211](https://github.com/astral-sh/ruff/pull/18211))
+- Gracefully handle salsa cancellations and panics in background request handlers ([#18254](https://github.com/astral-sh/ruff/pull/18254))
+
+### Playground
+
+- Better default settings for the playground ([#18316](https://github.com/astral-sh/ruff/pull/18316))
+- Use git-commit as ty playground version instead of 0.0.0 ([#18314](https://github.com/astral-sh/ruff/pull/18314))
+
+### Documentation
+
+- Add long help for `--config` argument ([#18285](https://github.com/astral-sh/ruff/pull/18285))
+- Add setup guide for using language server in Neovim ([#436](https://github.com/astral-sh/ty/pull/436))
+- Add warning that docs are autogenerated ([#18270](https://github.com/astral-sh/ruff/pull/18270))
+- Clarify `ty check` output default in documentation. ([#18246](https://github.com/astral-sh/ruff/pull/18246))
+- Document sources of autogenerated reference docs in contributor guide ([#490](https://github.com/astral-sh/ty/pull/490))
+- Fix anchor tag in `README` ToC ([#486](https://github.com/astral-sh/ty/pull/486))
+- Add type compendium ([#18263](https://github.com/astral-sh/ruff/pull/18263))
+- Add ToC and troubleshooting guidance to detailed README ([#464](https://github.com/astral-sh/ty/pull/464))
```

Comment:
Nit: We omit documentation changes most of the time for ruff (they're a bit noisy)

Reply:
Removed.